src/Input/InputLabel.js

```diff
@@ -108,7 +108,7 @@ function InputLabel(props: ProvidedProps & Props, context: { muiFormControl: Obj
   let shrink = shrinkProp;
 
   if (typeof shrink === 'undefined' && muiFormControl) {
-    shrink = muiFormControl.dirty || muiFormControl.focused;
+    shrink = muiFormControl.dirty || muiFormControl.focused || muiFormControl.adorned;
```

Comment:
Why force the control to have shrunken text?  In the case of an endAdornment specifically it seems to be to be more consistent to not force shrink.  I did some CSS-hacking to screenshot this: ![unshrunk text](https://i.gyazo.com/06b7bd6d128a275fe1c567c831dc8fda.png)